src/pages/index.js

```diff
@@ -18,81 +19,223 @@ import {
+  "Saving..."
 );
 
-section.renderItems();
+profileAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-// Modal Image
-const imagePreviewModal = new PopupWithImage(imageModalSelector);
-imagePreviewModal.setEventListeners();
+editAvatarPopup.setEventListeners();
+
+// Likes///
+function handleLikeClick(card) {
+  const isLiked = card.isLiked();
+
+  if (isLiked) {
+    api
+      .likeCountRemove(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error removing like:", error);
+      });
+  } else {
+    api
+      .likeCountAdd(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error adding like:", error);
+      });
+  }
+}
+
+const section = new Section({}, containerSelector);
 
 // Functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    (cardID) => handleDeleteClick(card, cardID),
+    handleLikeClick
+  );
+
+  return card.getView();
+}
+
 function addCard(data) {
   const cardData = {
     name: data.title,
     link: data.imageURL,
   };
 
-  renderCard(cardData);
-  addCardFormPopup.close();
+  // Show loading state
+  addNewCardButton.classList.add("loading");
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      const cardElement = renderCard(newCard);
+      section.addItem(cardElement);
+      addCardFormPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      // Hide loading state whether the operation succeeds or fails
+      addNewCardButton.classList.remove("loading");
+    });
 }
+const api = new Api(apiURL);
+
+const imagePreviewModal = new PopupWithImage(imageModalSelector);
+imagePreviewModal.setEventListeners();
+
+function handleCardImageClick(cardData) {
+  imagePreviewModal.open(cardData);
+}
+
+api
+  .getInitialCards()
+  .then((cardData) => {
+    cardData.forEach((cardItem) => {
+      const card = new Card(
+        cardItem,
+        "#card-template",
+        handleCardImageClick,
+        handleDeleteClick,
+        handleLikeClick
+      );
+
+      const cardElement = card.getView();
+
+      section.addItem(cardElement);
+    });
+  })
+  .catch((error) => {
+    console.error("Error fetching initial cards:", error);
+  });
 
 // card format
 const addCardModalSelector = "#add-card-modal";
-const data = () => {};
 
 const addCardFormPopup = new PopupWithForm(addCardModalSelector, addCard);
 addCardFormPopup.setEventListeners();
-addCardFormPopup.close();
+// addCardFormPopup.close();
 
 addNewCardButton.addEventListener("click", () => {
-  addCardFormValidator.disableButton(); // Disable the submit button
+  addCardFormValidator.disableButton();
 
+  // Start loading state here
+  addCardFormPopup.showLoading();
```

Comment:
Needs correction: Please remove this, there is no need to start the loading when you open the popup